src/utils/MediaEventHelper.ts

```diff
@@ -59,46 +83,52 @@ export class MediaEventHelper implements IDestroyable {
 
     private prepareSourceUrl = async () => {
         if (this.media.isEncrypted) {
-            const blob = await this.sourceBlob.value;
-            return URL.createObjectURL(blob);
+            const blob = await this.sourceTypedBlob.value;
+            if (blob.mimetype.startsWith("image/svg")) return createDataUrl(blob);
+            return URL.createObjectURL(blob.data);
         } else {
             return this.media.srcHttp;
         }
     };
 
     private prepareThumbnailUrl = async () => {
         if (this.media.isEncrypted) {
-            const blob = await this.thumbnailBlob.value;
+            const blob = await this.thumbnailTypedBlob.value;
             if (blob === null) return null;
-            return URL.createObjectURL(blob);
+            if (blob.mimetype.startsWith("image/svg")) return createDataUrl(blob);
+            return URL.createObjectURL(blob.data);
         } else {
             return this.media.thumbnailHttp;
         }
     };
 
-    private fetchSource = () => {
+    private fetchSource: () => Promise<ITypedBlob> = () => {
         if (this.media.isEncrypted) {
             const content = this.event.getContent<IMediaEventContent>();
-            return decryptFile(content.file, content.info);
+            return decryptFile(content.file, content.info).then(data => {
+                return { mimetype: content.info.mimetype, data };
+            });
         }
-        return this.media.downloadSource().then(r => r.blob());
+        return this.media.downloadSource().then(r => r.blob()).then(toTypedBlob);
     };
 
-    private fetchThumbnail = () => {
+    private fetchThumbnail: () => Promise<ITypedBlob | null> = () => {
         if (!this.media.hasThumbnail) return Promise.resolve(null);
 
         if (this.media.isEncrypted) {
             const content = this.event.getContent<IMediaEventContent>();
             if (content.info?.thumbnail_file) {
-                return decryptFile(content.info.thumbnail_file, content.info.thumbnail_info);
+                return decryptFile(content.info.thumbnail_file, content.info.thumbnail_info).then(data => {
+                    return { mimetype: content.info.thumbnail_info.mimetype, data };
```

Comment:
`info` might not exist, and `decryptFile` should be responsible for mimetype detection.

`thumbnail_info` also might not exist